CHANGELOG.md

```diff
@@ -75,6 +75,7 @@
 
 ### Behavior change
 - `Dataframe.columns` now returns native pandas Index object instead of Snowpark Index object.
+- Refactor and introduce `query_compiler` argument in `Index` constructor to create `Index` from query compiler.
```

Comment:
Users shouldn't be aware of the `query_compiler` argument to the `Index` constructor. Should we be noting this change at all in the release notes?

Reply:
why? the constructor is a public API though.

Reply:
It's a public API, but we also don't want or expect users to construct indexes out of query compilers at all.

I would consider this change a refactor that's not worth documenting, but I'm okay with leaving the note here.

Reply:
since it's a public API, it's safer to document any changes related to it. I agree that we won't find any user to be affected by this.